core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1918,3 +1980,19 @@ object RDD {
     new DoubleRDDFunctions(rdd.map(x => num.toDouble(x)))
   }
 }
+
+/**
+ * The random level of RDD's output (i.e. what `RDD#compute` returns), which indicates how the
+ * output will diff when Spark reruns the tasks for the RDD. There are 3 random levels, ordered
+ * by the randomness from low to high:
+ * 1. IDEMPOTENT: The RDD output is always same (including order) when rerun.
```

Comment:
here too, idempotent is the wrong word for this ... deteminstic?  partition-ordered? (I guess "ordered" could make it seem like the entire data is ordered ...)